src/Typewriter/MetadataPreprocessor.cs

```diff
@@ -181,5 +200,94 @@ internal static void AddLongDescriptionToThumbnail()
+        internal static void ReorderElements(MetadataDefinitionType metadataDefinitionType, 
+                                             string targetGlobalElementName, 
+                                             List<string> newElementOrder,
+                                             string bindingParameterType = "")
+        {
+            // Actions or Functions require a binding element.
+            if (String.IsNullOrEmpty(bindingParameterType) && 
+                metadataDefinitionType == MetadataDefinitionType.Action || 
+                metadataDefinitionType == MetadataDefinitionType.Function)
+            {
+                throw new ArgumentNullException(nameof(bindingParameterType), "The binding parameter type must be set in case an Action" +
+                    " or Function with the same name and parameter list.");
+            }
+            
+            // Validate that the specified new element order is meaningful.
+            if (newElementOrder.Count < 2)
+            {
+                throw new ArgumentOutOfRangeException(nameof(newElementOrder), "ReorderElements: expected 2 or more elements to reorder.");
+            }
+
+            // Sort the newElementOrder so we can compare the sequence to the potential overloads returned for the
+            // the targetGlobalElementName. We should only ever find a single match.
+            var newElementsAlphaOrdered = newElementOrder.OrderByDescending(x => x.ToString());
+
+            try
+            {
+                // Get the target element that has the target type (i.e. Action, EntityType), with the target Name (i.e. forward)
+                // where it has the same element list as the new element list 
+                // where its binding parameter (the first element) has a Type attribute that matches the given 
+                // bindingParameterType in the case of Action or Function.
+
+                var results = xMetadata.Descendants()
+                    .Where(x => x.Name.LocalName == metadataDefinitionType.ToString())
+                    .Where(x => x.Attribute("Name").Value == targetGlobalElementName)
+                    .Where(el => el.Elements().Select(a => a.Attribute("Name").Value)
+                                              .OrderByDescending(e => e.ToString())
+                                              .SequenceEqual(newElementsAlphaOrdered));
+
+                XElement targetElement = null;
+
+                // Reordering elements by element Name attributes. Useful for non Action or Function.
+                if (String.IsNullOrEmpty(bindingParameterType))
+                {
+                    targetElement = results.FirstOrDefault();
+                }
+                else // We are reordering an Action or Function and must match the bindingParameterType.
+                {
+                    targetElement = results.Where(e => e.Elements().Any(a => a.Attribute("Type").Value == bindingParameterType)).FirstOrDefault();
```

Comment:
Here, we are checking if any of the parameters/elements has the same type as the `bindingParameterType` . 
If the bindingParameter is not really of the same type as `bindingParameterType` but one of the other params has that type we might end up with a false positive.
Since the bindingParameter is typically the first parameter, would it be better to check if the type of the **first** element type matches the `bindingParameterType`. I think that may be a better invariant.

Reply:
Good catch!!! The bindingParameter is **always** the first parameter. I will fix.

Reply:
@andrueastman Made the suggested fix.